client/scripts/views/pages/subscriptions/chainEventSettings.ts

```diff
@@ -72,12 +71,48 @@ const EventSubscriptionRow: m.Component<IEventSubscriptionRowAttrs, {}> = {
   }
 };
 
+const IndividualEventSubscriptions: m.Component<{}, {
+  chain: string;
+  eventKinds: IChainEventKind[];
+  allSupportedChains: string[];
+  isSubscribedAll: boolean;
+  isEmailAll: boolean;
+}> = {
+  view: (vnode) => {
+    let titler;
+    if (vnode.state.chain === ChainNetwork.Edgeware || vnode.state.chain === 'edgeware-local') {
+      titler = SubstrateEvents.Title;
+      vnode.state.eventKinds = SubstrateTypes.EventKinds;
```

Comment:
why are we reassigning to `state` every time `view` runs?

Reply:
I reorganized this component to be more sensible, but like I said below, it was a specific chain-event testing component that isn't used in production (as we batch the subscriptions into types: 'democracy'/'treasury' etc)